base/genericmemory.jl

```diff
@@ -238,12 +238,13 @@ getindex(A::Memory, c::Colon) = copy(A)
 function setindex!(A::Memory{T}, x, i1::Int) where {T}
     val = x isa T ? x : convert(T,x)::T
     ref = memoryrefnew(memoryref(A), i1, @_boundscheck)
-    memoryrefset!(ref, val, :not_atomic, @_boundscheck)
+    # boundscheck emitted by `memoryrefnew` also checks the index for memoryrefset!
+    memoryrefset!(ref, val, :not_atomic, false)
     return A
 end
 
 function setindex!(A::Memory{T}, x, i1::Int, i2::Int, I::Int...) where {T}
-    @inline
+    @_propagate_inbounds_meta
```

Comment:
Maybe this is a silly question, but does the `@inbounds if destp < srcp || destp > endp` above (in `unsafe_copyto!`) apply `@inbounds` to the `else` branch, too? Also, shouldn't the inbounds/prop inbounds pattern be applied to `getindex`, too?